src/vmm/src/arch/aarch64/vcpu.rs

```diff
@@ -237,7 +257,32 @@ mod tests {
         vm.get_preferred_target(&mut kvi).unwrap();
         vcpu.vcpu_init(&kvi).unwrap();
 
-        setup_boot_regs(&vcpu, 0, 0x0, &mem).unwrap();
+        setup_boot_regs(&vcpu, 0, 0x0, &mem, &optional_capabilities).unwrap();
+
+        // Check that the register is reset on compatible kernels.
+        // Because there is a delta in time between we reset the register and time we
+        // read it, we cannot compare with 0. Instead we read CNTVNT_EL0 (Virtual counter)
+        // from the host. The host value should be much bigger than the VM physical counter.
+        if optional_capabilities.counter_offset {
+            let mut reg_bytes = [0_u8; 8];
+            vcpu.get_one_reg(SYS_CNTPCT_EL0, &mut reg_bytes).unwrap();
+            let counter_value = u64::from_le_bytes(reg_bytes);
+
+            let virt_count = unsafe {
+                // Rust is so smart, it cannot figure out that `virt_count`
+                // is read later in the assert!, so it complains about unused assignment.
```

Comment:
nit: I mean, the assignment is unused. The assembly overwrites the value again, so there should be no need to initialize this variable, right?

Reply:
fixed